### PR TITLE
modernify cmake

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -73,18 +76,20 @@ if(BUILD_TESTS)
...
-include_directories(${xtensor_INCLUDE_DIRS})
-
 # Installation
 # ============
 
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
+# install(FILES ${XTENSOR_IO_HEADERS}
+#         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/xtensor-io)
```
Comment (SylvainCorlay) [inline CMakeLists.txt]: left-over comment?

**Comment by SylvainCorlay** [comment]
This look ok to me. Is this ready to go?

**Comment by wolfv** [comment]
It's getting closer to being ready.
In short, we'll also need to ship the two cmake modules in the `modules/` folder. (Should we install them with CMake?

Also, we should probably add cmake options to enable the different libraries? What do you think?

Additionally, documenting how/why setting RPATH in CMake works should be nice for our users who try to use xtensor-io from conda.

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,17 +53,45 @@ set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${CMAKE_SOURCE_DIR}/modules")
...
+
+add_library(xtensor-io INTERFACE)
+target_include_directories(xtensor-io
+  INTERFACE
+    $<BUILD_INTERFACE:${XTENSOR_IO_INCLUDE_DIR}>
+    # LIBRARIES
+    $<BUILD_INTERFACE:${OIIO_INCLUDE_DIRS}>
+    $<BUILD_INTERFACE:${LIBSNDFILE_INCLUDE_DIRS}>
+    $<BUILD_INTERFACE:${ZLIB_INCLUDE_DIRS}>
+    # INSTALLATIOn
```
Comment (SylvainCorlay) [inline CMakeLists.txt]: small typo

File: modules/FindOIIO.cmake [hunk 3]
```diff
@@ -22,20 +22,25 @@
 # language governing permissions and limitations under the Apache License.
 #
 
+set(_OIIO_HINTS
+  "${CMAKE_INSTALL_PREFIX}"
+  "${OIIO_LOCATION}"
+  "$ENV{OIIO_LOCATION}"
+  "${OIIO_BASE_DIR}"
+  "/opt/oiio"
```
Comment (SylvainCorlay) [inline modules/FindOIIO.cmake]: :+1:

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -124,16 +102,56 @@ if(DOWNLOAD_GTEST OR GTEST_SRC_DIR)
...
+    ${CMAKE_DL_LIBS}
+    ${GTEST_BOTH_LIBRARIES}
+    ${CMAKE_THREAD_LIBS_INIT}
 )
 
 add_custom_target(xtest
     COMMAND test_xtensor_io
     DEPENDS ${XTENSOR_IO_TARGET})
 
+# Get all propreties that cmake supports
```
Comment (SylvainCorlay) [inline test/CMakeLists.txt]: Do you want to keep this in the cmake?

**Pull request merged**